x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type.ts

```diff
@@ -159,6 +162,26 @@ export function getAlertType(
         { name: 'index', description: actionVariableContextIndexLabel },
       ],
     },
+    useSavedObjectReferences: {
+      extractReferences: (params) => {
+        const [searchConfiguration, references] = extractReferences(params.searchConfiguration);
+
+        if (isEsQueryAlert(params.searchType)) {
+          return { params: params as EsQueryAlertParamsExtractedParams, references };
+        }
+        const newParams = { ...params, searchConfiguration } as EsQueryAlertParamsExtractedParams;
+        return { params: newParams, references };
```

Comment:
Since searchConfiguration is a `searchSource` only param, extracting references will always return an empty array, right? So we don't need to execute `extractReferences`
```suggestion
        if (isEsQueryAlert(params.searchType)) {
          return { params: params as EsQueryAlertParamsExtractedParams, references: [] };
        }
        const [searchConfiguration, references] = extractReferences(params.searchConfiguration);
        const newParams = { ...params, searchConfiguration } as EsQueryAlertParamsExtractedParams;
        return { params: newParams, references };
```